x-pack/test/apm_api_integration/tests/transactions/error_rate.spec.ts

```diff
@@ -91,149 +77,185 @@ export default function ApiTest({ getService }: FtrProviderContext) {
+        return timerange(start, end)
+          .interval(interval)
+          .rate(transaction.successRate)
+          .generator((timestamp) =>
+            serviceGoProdInstance
+              .transaction({ transactionName: transaction.name })
+              .timestamp(timestamp)
+              .duration(1000)
+              .success()
+          )
+          .merge(
+            timerange(start, end)
+              .interval(interval)
+              .rate(transaction.failureRate)
+              .generator((timestamp) =>
+                serviceGoProdInstance
+                  .transaction({ transactionName: transaction.name })
+                  .errors(
+                    serviceGoProdInstance
+                      .error({
+                        message: 'Error 1',
+                        type: transaction.name,
+                        groupingName: 'Error test',
+                      })
+                      .timestamp(timestamp)
+                  )
+                  .duration(1000)
+                  .timestamp(timestamp)
+                  .failure()
+              )
+          );
+      });
 
-        it('returns some data', () => {
-          expect(errorRateResponse.currentPeriod.average).to.be.greaterThan(0);
-          expect(errorRateResponse.previousPeriod.average).to.be(null);
+      await synthtraceEsClient.index(documents);
+    });
 
-          expect(errorRateResponse.currentPeriod.timeseries.length).to.be.greaterThan(0);
-          expect(errorRateResponse.previousPeriod.timeseries).to.empty();
+    after(() => synthtraceEsClient.clean());
 
-          const nonNullDataPoints = errorRateResponse.currentPeriod.timeseries.filter(
-            ({ y }) => y !== null
-          );
+    describe('returns the transaction error rate', () => {
+      let errorRateResponse: ErrorRate;
 
-          expect(nonNullDataPoints.length).to.be.greaterThan(0);
+      before(async () => {
+        const response = await fetchErrorCharts({
+          query: { transactionName: config.firstTransaction.name },
         });
+        errorRateResponse = response.body;
+      });
 
-        it('has the correct start date', () => {
-          expectSnapshot(
-            new Date(first(errorRateResponse.currentPeriod.timeseries)?.x ?? NaN).toISOString()
-          ).toMatchInline(`"2021-08-03T06:50:00.000Z"`);
-        });
+      it('returns some data', () => {
+        expect(errorRateResponse.currentPeriod.average).to.be.greaterThan(0);
+        expect(errorRateResponse.previousPeriod.average).to.be(null);
 
-        it('has the correct end date', () => {
-          expectSnapshot(
-            new Date(last(errorRateResponse.currentPeriod.timeseries)?.x ?? NaN).toISOString()
-          ).toMatchInline(`"2021-08-03T07:20:00.000Z"`);
-        });
+        expect(errorRateResponse.currentPeriod.timeseries.length).to.be.greaterThan(0);
+        expect(errorRateResponse.previousPeriod.timeseries).to.empty();
 
```

Comment:
This code was based on snapshots which in turn came from es archives. We wanted it to be easy to update the es archives automatically without having to change the code, and the tests were therefore somewhat vague.

I don't think we need to stick to this approach, and you should be able to assert more specifically:

```suggestion
        expect(errorRateResponse.currentPeriod.timeseries.length).to.be(33);
        expect(errorRateResponse.previousPeriod.timeseries).to.empty();

```